### DISCUSSION TITLE
Identifiably sortable columns?

### DISCUSSION THREAD

**Discussion by RalphMack** [6pac/SlickGrid#988] [Q&A] [answered]
In old SlickGrid, if a column was sortable but not being sorted, it had the double arrows (kind of like a diamond) on it that announced it was sortable. Columns that were not sortable had no arrows. What is the best way to restore this behavior? This is the one remaining thing that customers would find confusing enough to balk us from deploying our app with our changes to use the new version.

The remaining question is more a matter of UX optimization:

In the old SlickGrid, when you sorted on it, it flipped to up then down then you had to do a Control-click (or Command-click) to get it back to double arrows, which I didn't care for much. I much prefer the tristate up-down-none click, holding down a shift to affect (in a tri-state manner) just one column of a multi-column sort.  Clicking to sort another column without a shift would remove the existing sort. It's seems to require the least to remember - just one modifier key involved - and hence be the most discoverable without help (because real heroes never ask for help ;) ). 

I've tried a couple of the sort options, but I couldn't find documentation, and they all seem a little quirky. I want each of the sort indicators to be tristate and I want multicolumn sorting, but tristateMultiColumnSort is just confusing. (Maybe if there were a tristateSort option separate from multiColumnSort and you could select both?) I haven't come across an option that does something like what I want, but maybe I missed it. 

For both of these matters, what option would I use or where would I apply customizations to get the desired behavior?

**Comment by ghiscoding** [comment]
in Slickgrid-Universal, I use these options to get the behavior and UX that most users are used to, you can try any of the Slickgrid-Universal [live examples](https://ghiscoding.github.io/slickgrid-universal/) to test it out.
```ts
  multiColumnSort: true,
  numberedMultiColumnSort: true,
  tristateMultiColumnSort: false,
  sortColNumberInSeparateSpan: true,
```
I've never shown both up/down caret at the same time and I'm not at all interested in showing it either, all my grids are sortable and all users know and that's it. Without looking at the code, I think that the div holding the sort icon is added dynamically and also removed when sorting is neutral, so I'm not sure how you would add an icon on a div that no longer exists...

I'm personally only interested in the behavior described above so I don't know how to change to for other behavior/UX, you'll have to play with all available examples to see which one is close to what you want

**Comment by 6pac** [comment, marked as answer]
Just another comment, I checked the old original MLeibman repo and early versions of this one, and we've never had the double diamond column icon as far as I can see. That was likely a customisation of your copy of the grid.  
There are a number of examples where you can view the UI for different sorting styles.

**Reply by RalphMack** [reply]
Who did what when is probably lost in the mists of time now, across three source control systems by now on our end - we've been using it for a decade - thanks for doing the research. I'll mark yours as the answer.
